internal/app/referenceserver/checks.go

```diff
@@ -145,12 +146,15 @@ func checkCodec(expected v1.Codec, req *http.Request, feedback *feedbackWriter)
 		feedback.Printf("invalid expected codec %d", expected)
 		return
 	}
-	contentType, hasContentType := getHeader(req.Header, "content-type", feedback)
+	contentType, _ := getHeader(req.Header, "content-type", feedback)
 	var actual string
 	switch {
 	case req.Method == http.MethodGet:
-		if hasContentType {
-			feedback.Printf("content-type header should not appear with method GET")
+		// Servers should test for an empty request body by attempting a read.
+		// If no body is present, it should return an immediate EOF.
```

Comment:
I think we want to keep the `hasContentType` var above and then have something like so here:
```
// TODO: Require that no content-type be present. Must wait for
//       https://github.com/connectrpc/connect-go/pull/644 to
//       be release and incorporated into the reference client.
_ = hasContentType
```